dotCMS/src/main/java/com/dotmarketing/servlets/ShortyServlet.java

```diff
@@ -1,188 +1,284 @@
 
+          dispatch.forward(request, response);
+      }
+  }
 
-    if(isImage){
-      path += "/filter/";
-      path += (w+h > 0) ? "Resize," : "";
-      path += (jpeg) ? "Jpeg/jpeg_q/75" : "";
-      path += (jpeg && jpegp) ? "/jpeg_p/1" : "";
-      path += (w > 0) ? "/resize_w/" + w : "";
-      path += (h > 0) ? "/resize_h/" + h : "";
-    }
+  private void addImagePath(final int weight,
+                            final int height,
+                            final boolean jpeg,
+                            final boolean jpegp,
+                            final boolean isImage,
+                            final StringBuilder pathBuilder) {
+      if(isImage) {
+
+          pathBuilder.append("/filter/");
+          pathBuilder.append(weight+height > 0? "Resize,"      : StringPool.BLANK);
+          pathBuilder.append(jpeg? "Jpeg/jpeg_q/75"            : StringPool.BLANK);
+          pathBuilder.append(jpeg && jpegp? "/jpeg_p/1"        : StringPool.BLANK);
+          pathBuilder.append(weight > 0? "/resize_w/" + weight : StringPool.BLANK);
+          pathBuilder.append(height > 0? "/resize_h/" + height : StringPool.BLANK);
+      }
+  }
 
-    request.setAttribute(SHORTY_SERVLET_FORWARD_PATH, path);
-    
-    
-    
-    RequestDispatcher dispatch = request.getRequestDispatcher(path);
-    if(dispatch!=null){
-      dispatch.forward(request, response);
+  private void addHeaders(final HttpServletResponse response, final boolean live) {
+
+    if (!live) {
+      response.setHeader("Pragma", "no-cache");
+      response.setHeader("Cache-Control", "no-cache");
+      response.setDateHeader("Expires", 0);
     }
   }
 
+  private boolean isValidRequest(final HttpServletRequest request,
+                                 final HttpServletResponse response, PageMode mode) throws DotDataException, DotSecurityException, PortalException, SystemException, IOException {
+
+    final Host host     = this.hostWebAPI.getCurrentHost(request);
+    // Checking if host is active
+    if (!mode.isAdmin && !this.versionableAPI.hasLiveVersion(host)) {
+
+      response.sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE,
+          LanguageUtil.get("server-unavailable-error-message"));
+      return false;
+    }
 
+    return true;
+  }
 
-    protected final String inodePath(final Contentlet con, final String tryField, final boolean live)
+
+  protected final String inodePath(final Contentlet contentlet,
+                                   final String tryField,
+                                   final boolean live)
             throws DotStateException, DotDataException {
 
-        final Optional<Field> fieldOpt = resolveField(con, tryField);
+        final Optional<Field> fieldOpt = resolveField(contentlet, tryField);
 
         if (!fieldOpt.isPresent()) {
-            return "/" + con.getInode() + "/" + FILE_ASSEST_DEFAULT;
+            return new StringBuilder(StringPool.FORWARD_SLASH).append(contentlet.getInode())
```

Comment:
This is unnecessary optimization - the compiler takes care of string concatenation and we lose readability

Reply:
done